src/github_runner_image_builder/cloud_image.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module for downloading images from cloud-images.ubuntu.com."""
+
+# gzip needs to be preloaded to extract github runner tar.gz. This is because within the chroot
+# env, tarfile module tries to import gzip dynamically and fails.
+import gzip  # noqa: F401 # pylint: disable=unused-import
+import hashlib
+import logging
+import typing
+import urllib.error
+import urllib.request
+import urllib.response
+from pathlib import Path
+
+import requests
+
+from github_runner_image_builder.config import Arch, BaseImage
+from github_runner_image_builder.errors import BaseImageDownloadError, UnsupportedArchitectureError
+from github_runner_image_builder.utils import retry
+
+logger = logging.getLogger(__name__)
+
+SupportedBaseImageArch = typing.Literal["amd64", "arm64"]
```

Comment:
This actually duplicates the logic from `_get_supported_runner_arch`. you are validating in that method and also with pydantic

Reply:
This is just to help with type hinting so that it becomes easier to code with intellisense :D